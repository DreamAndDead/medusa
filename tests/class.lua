local Point = class(function(Point)
    local function __init__(kvs, self, x, y)
        self = get_posarg(kvs, 'self', self, nil, '__init__')
        x = get_posarg(kvs, 'x', x, nil, '__init__')
        y = get_posarg(kvs, 'y', y, nil, '__init__')
        self.x = x
        self.y = y
    end
    local function get_x(kvs, self)
        self = get_posarg(kvs, 'self', self, nil, 'get_x')
        return self.x
    end
    local function get_y(kvs, self)
        self = get_posarg(kvs, 'self', self, nil, 'get_y')
        return self.y
    end
    Point.__init__ = __init__
    Point.get_x = get_x
    Point.get_y = get_y
    return Point
end, {}, "Point")
local p = Point(merge_kwargs({}, {}), 5, 10)
assert(bool((p.get_x(merge_kwargs({}, {})) == 5)))
assert(bool((p.get_y(merge_kwargs({}, {})) == 10)))
assert(bool((p.x == 5)))
assert(bool((p.y == 10)))
local Dog = class(function(Dog)
    local species = "mammal"
    local function __init__(kvs, self, name, age)
        self = get_posarg(kvs, 'self', self, nil, '__init__')
        name = get_posarg(kvs, 'name', name, nil, '__init__')
        age = get_posarg(kvs, 'age', age, nil, '__init__')
        self.name = name
        self.age = age
    end
    Dog.species = species
    Dog.__init__ = __init__
    return Dog
end, {}, "Dog")
local d = Dog(merge_kwargs({}, {}), "roster", 18)
assert(bool((Dog.species == "mammal")))
assert(bool((d.name == "roster")))
assert(bool((d.age == 18)))
local Person = class(function(Person)
    local function __init__(kvs, self, name, age)
        self = get_posarg(kvs, 'self', self, nil, '__init__')
        name = get_posarg(kvs, 'name', name, nil, '__init__')
        age = get_posarg(kvs, 'age', age, nil, '__init__')
        self.name = name
        self.age = age
    end
    local function talk(kvs, self)
        self = get_posarg(kvs, 'self', self, nil, 'talk')
        return "person talks"
    end
    Person.__init__ = __init__
    Person.talk = talk
    return Person
end, {}, "Person")
local Chinese = class(function(Chinese)
    local function __init__(kvs, self, name, age, country)
        self = get_posarg(kvs, 'self', self, nil, '__init__')
        name = get_posarg(kvs, 'name', name, nil, '__init__')
        age = get_posarg(kvs, 'age', age, nil, '__init__')
        country = get_posarg(kvs, 'country', country, nil, '__init__')
        Person.__init__(merge_kwargs({}, {}), self, name, age)
        self.country = country
    end
    local function walk(kvs, self)
        self = get_posarg(kvs, 'self', self, nil, 'walk')
        return "chinese walks"
    end
    Chinese.__init__ = __init__
    Chinese.walk = walk
    return Chinese
end, {Person}, "Chinese")
local American = class(function(American)
    local function talk(kvs, self)
        self = get_posarg(kvs, 'self', self, nil, 'talk')
        return "I love talk show"
    end
    American.talk = talk
    return American
end, {Person}, "American")
local c = Chinese(merge_kwargs({}, {}), "john", 18, "China")
assert(bool((c.name == "john")))
assert(bool((c.age == 18)))
assert(bool((c.country == "China")))
assert(bool((c.talk(merge_kwargs({}, {})) == "person talks")))
assert(bool((c.walk(merge_kwargs({}, {})) == "chinese walks")))
local a = American(merge_kwargs({}, {}), "jane", 24)
assert(bool((a.name == "jane")))
assert(bool((a.age == 24)))
assert(bool((a.talk(merge_kwargs({}, {})) == "I love talk show")))
return {
    Point = Point,
    p = p,
    Dog = Dog,
    d = d,
    Person = Person,
    Chinese = Chinese,
    American = American,
    c = c,
    a = a,
}
