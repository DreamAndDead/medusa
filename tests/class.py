class Animal:
    PLANET = "Earth"

    def __init__(self, name):
        self.name = name

    def say_hello(self):
        print("Hello, my name is: " + self.name + "!")


class Dog(Animal):
    def say_hello(self):
        print("Hello, my name is: " + self.name + "!")
        self.bark()

    def bark(self):
        print("Bark! Bark! Bark!")


sparky = Animal("Sparky")
barky = Dog("Barky")

sparky.say_hello()
barky.say_hello()

barky.bark()

print("Animal.PLANET = ", Animal.PLANET)
print("sparky.PLANET = ", sparky.PLANET)
print("barky.PLANET = ", barky.PLANET)

Animal.PLANET = "Mars"

print("Animal.PLANET = ", Animal.PLANET)
print("sparky.PLANET = ", sparky.PLANET)