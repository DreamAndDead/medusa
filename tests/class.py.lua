dofile('./pythonlua/luainit.lua')

local Animal = class(function(Animal)
    Animal.PLANET = "Earth"
    function Animal.__init__(self, name)
        self.name = name
    end
    function Animal.say_hello(self)
        print((("Hello, my name is: " + self.name) + "!"))
    end
    return Animal
end, {})
local Dog = class(function(Dog)
    function Dog.say_hello(self)
        print((("Hello, my name is: " + self.name) + "!"))
        self.bark()
    end
    function Dog.bark(self)
        print("Bark! Bark! Bark!")
    end
    return Dog
end, {Animal})
local sparky = Animal("Sparky")
local barky = Dog("Barky")
sparky.say_hello()
barky.say_hello()
barky.bark()
print("Animal.PLANET = ", Animal.PLANET)
print("sparky.PLANET = ", sparky.PLANET)
print("barky.PLANET = ", barky.PLANET)
Animal.PLANET = "Mars"
print("Animal.PLANET = ", Animal.PLANET)
print("sparky.PLANET = ", sparky.PLANET)