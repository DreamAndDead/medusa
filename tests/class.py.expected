Hello, my name is: Sparky!
Hello, my name is: Barky!
Bark! Bark! Bark!
Bark! Bark! Bark!
Animal.PLANET = 	Earth
sparky.PLANET = 	Earth
barky.PLANET = 	Earth
Animal.PLANET = 	Mars
sparky.PLANET = 	Mars
